src/tls.c

```diff
@@ -7150,7 +7183,120 @@ static void findEccPqc(int *ecc, int *pqc, int group)
  * kse   The key share entry object.
  * returns 0 on success, otherwise failure.
  */
-#ifdef HAVE_LIBOQS
+#ifdef WOLFSSL_HAVE_KYBER
+static int TLSX_KeyShare_GenPqcKey(WOLFSSL *ssl, KeyShareEntry* kse)
+{
+    int ret = 0;
+    int type = 0;
+    KyberKey kem[1];
+    byte* pubKey = NULL;
+    byte* privKey = NULL;
+    KeyShareEntry *ecc_kse = NULL;
+    int oqs_group = 0;
+    int ecc_group = 0;
+    word32 privSz = 0;
+    word32 pubSz = 0;
+
+    findEccPqc(&ecc_group, &oqs_group, kse->group);
+    ret = kyber_id2type(oqs_group, &type);
+    if (ret == NOT_COMPILED_IN) {
+        WOLFSSL_MSG("Invalid Kyber algorithm specified.");
+        ret = BAD_FUNC_ARG;
+    }
+
+    if (ret == 0) {
+        ret = wc_KyberKey_Init(type, kem, ssl->heap, INVALID_DEVID);
```

Comment:
Lets use the devID here, either from a dereference of 'ssl' or a function call to ` wolfSSL_CTX_GetDevId(WOLFSSL_CTX* ctx, WOLFSSL* ssl);`

Reply:
Fixed to match what we do in TLSX_KeyShare_ProcessDh().